charon-ml/src/GAst.ml

```diff
@@ -51,13 +51,15 @@ and any_decl_id =
 class ['self] iter_statement_base_base =
   object (self : 'self)
     inherit [_] iter_rvalue
-    method visit_abort_kind : 'env -> abort_kind -> unit = fun _ _ -> ()
+    method visit_abort_kind : 'env. 'env -> abort_kind -> unit = fun _ _ -> ()
```

Comment:
You're changing `GAst.ml`, which as explained at the top is automatically-generated. You need to change `charon/src/bin/generate-ml/templates/GAst.ml` instead and run `make generate-ml`.

Reply:
Actually this particular visitor is automatically-generated, so you'll have to update the visitor-generating code in `charon/src/bin/generate-ml/main.rs`, specifically `fn generate_visitor_bases`.

Reply:
Yes, I'm perfectly aware of that.
I'm currently fighting with a weirdness of OCaml: it seems that because this visitor has only one method, OCaml doesn't manage to see that the `'env` type variable here should be the same as in the methods inherited from the `rvalue` visitors. It leads to an error when type-checking `st_substitute_visitor`.

Reply:
The solution is to move the `abort_kind` visit methods to the ULLBC visitors, which actually makes more sense (cf the commit I just pushed).